### PR TITLE
downloadIstioCandidate.sh supports passing env TARGET_OS

### PR DESCRIPTION
**Pull request by YonkaFang**
**Please provide a description of this PR:**

Sometimes we may download Istio on a personal/work computer (Mac) and then transfer it to a test environment (Linux) for installation, in which case we need to specify the target OS.

### CODE REVIEW

File: release/downloadIstioCandidate.sh [hunk 1]
```diff
@@ -21,13 +21,13 @@
 # The script fetches the latest Istio release candidate and untars it.
 # You can pass variables on the command line to download a specific version
 # or to override the processor architecture. For example, to download
-# Istio 1.6.8 for the x86_64 architecture,
-# run curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.6.8 TARGET_ARCH=x86_64 sh -.
+# Istio 1.6.8 for the x86_64 architecture and linux OS,
+# run curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.6.8 TARGET_ARCH=x86_64 TARGET_OS=Linux sh -.
```
Comment (zirain) [inline release/downloadIstioCandidate.sh:25]: from L34, this should be `linux`?

Comment (YonkaFang) [reply release/downloadIstioCandidate.sh:25]: In fact, Istio can be downloaded correctly here using either "linux" or "Linux". However, considering the difference between TARGET_ARCH and ISTIO_ARCH, the TARGET_OS selection corresponds to the output of uname, while OSEXT in the original text is used together with ISTIO_ARCH to concatenate the file name.

At the same time, this is consistent with the logic at https://github.com/istio/istio/blob/f0f295a2c2366a91ee9c19ffc11326c1a8f48488/release/downloadIstioCandidate.sh#L105 .

**Comment by ericvn** [comment]
I would add the same logic in the downloadIstioCtl.sh to keep them consistent.

**Comment by YonkaFang** [comment]
> I would add the same logic in the downloadIstioCtl.sh to keep them consistent.

@ericvn done

**Comment by ericvn** [comment]
This code does not work as you desire. You can try cd'ing to the release directory and run `cat downloadIstioCandidate.sh | ISTIO_VERSION=1.18.0 TARGET_ARCH=x86_64 sh -` fr example and this does not work. You'll see:
```
Unable to download Istio 1.18.0 at this moment!
Please verify the version you are trying to download.
```
A follow-on test would be to add TARGET_OS into the command and verifies that it pulls the correct version as well.

As an aside, you might update the 1.6.8 in the remarks to be a more recent version.

File: release/downloadIstioCandidate.sh [hunk 2]
```diff
@@ -21,13 +21,13 @@
...
 
 set -e
 
 # Determines the operating system.
-OS="$(uname)"
+OS="{TARGET_OS:-$(uname)}"
```
Comment (ericvn) [inline release/downloadIstioCandidate.sh:30]: Giving you a way to test before you merge, but here is what is likely the fix.
```suggestion
OS="${TARGET_OS:-$(uname)}"
```

Comment (ericvn) [reply release/downloadIstioCandidate.sh:30]: Actually if I specify `TARGET_OS=Linux` (I had specified a lower case l in my earlier testing), it seems to work.

Comment (ericvn) [reply release/downloadIstioCandidate.sh:30]: Looking we don't seem to do any validation of the TARGET_ARCH either.

Comment (YonkaFang) [reply release/downloadIstioCandidate.sh]: Yes, this script does not strictly handle the values of OS/ARCH. The reason I used "Linux" here is to make the following "ARCH_SUPPORTED" section of code work.

**Comment by YonkaFang** [comment]
> This code does not work as you desire. You can try cd'ing to the release directory and run `cat downloadIstioCandidate.sh | ISTIO_VERSION=1.18.0 TARGET_ARCH=x86_64 sh -` fr example and this does not work. You'll see:
> 
> ```
> Unable to download Istio 1.18.0 at this moment!
> Please verify the version you are trying to download.
> ```
> 
> A follow-on test would be to add TARGET_OS into the command and verifies that it pulls the correct version as well.
> 
> As an aside, you might update the 1.6.8 in the remarks to be a more recent version.

@ericvn 

I've done the test as below:

```sh
 cat release/downloadIstioCandidate.sh | ISTIO_VERSION=1.18.0 TARGET_ARCH=x86_64  sh -               

Downloading istio-1.18.0 from https://github.com/istio/istio/releases/download/1.18.0/istio-1.18.0-osx.tar.gz ...


cat release/downloadIstioCandidate.sh | ISTIO_VERSION=1.18.0 TARGET_ARCH=x86_64 TARGET_OS=linux sh -


Unable to download Istio 1.18.0 at this moment!
Please verify the version you are trying to download.

cat release/downloadIstioCandidate.sh | ISTIO_VERSION=1.18.0 TARGET_ARCH=x86_64 TARGET_OS=Linux sh -

Downloading istio-1.18.0 from https://github.com/istio/istio/releases/download/1.18.0/istio-1.18.0-linux-amd64.tar.gz ...
```

I think that this result is as expected.

File: release/downloadIstioCandidate.sh (hunk 2 above)
Comment (ericvn) [reply release/downloadIstioCandidate.sh]: name does return `Linux` on a Ubuntu VM and expected on others.

**Comment by ericvn** [comment]
I ran some tests post merge, like `curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.18.0 TARGET_OS=Darwin sh -` to verify function and the old commands from istio.io work correctly.

**Pull request merged**